--- gmp-4.1.3/configure.in.orig	2004-06-24 04:26:23.000000000 +0200
+++ gmp-4.1.3/configure.in	2004-06-24 04:42:25.095089576 +0200
@@ -886,6 +886,12 @@
             SPEED_CYCLECOUNTER_OBJ_64=sparcv9.lo
             cyclecounter_size_64=2
             ;;
+          sparc64-pld-linux*)
+            abilist="64 32"
+            path_64="sparc64"
+            SPEED_CYCLECOUNTER_OBJ_64=sparcv9.lo
+            cyclecounter_size_64=2
+            ;;
           *-*-*bsd*)
             # NetBSD and OpenBSD sparc64
             abilist="64"
