--- gmp-3.1.1/configure.in.orig	Fri Aug 10 21:15:48 2001
+++ gmp-3.1.1/configure.in	Fri Aug 10 21:27:06 2001
@@ -1,5 +1,5 @@
 dnl  Process this file with autoconf to produce a configure script.
-
+m4_pattern_allow([^dnl$])
 
 dnl  Copyright (C) 1996, 1997, 1998, 1999, 2000 Free Software Foundation, Inc.
 dnl
@@ -36,6 +36,8 @@
 dnl  GMP specific
 GMP_INIT(config.m4)
 
+AC_PROG_CC
+
 AC_ARG_ENABLE(assert,
 AC_HELP_STRING([--enable-assert],[enable ASSERT checking [default=no]]),
 [case "${enableval}" in
@@ -336,46 +336,6 @@
     eval CFLAGS=\$gmp_cflags_$CC  
   fi
 
-  # Try compiler flags that may work with only some compiler versions.
-  # gmp_optcflags: All or nothing.
-  eval optcflags=\$gmp_optcflags_$CC
-  if test -n "$optcflags"; then
-    CFLAGS_save="$CFLAGS"
-    CFLAGS="$CFLAGS $optcflags"
-    AC_MSG_CHECKING([whether $CC accepts $optcflags])
-    AC_LANG_C
-    AC_TRY_COMPILER([int main(){return(0);}], optok, cross)
-    if test "$optok" = "yes"; then
-      AC_MSG_RESULT([yes])
-    else
-      AC_MSG_RESULT([no])
-      CFLAGS="$CFLAGS_save"
-    fi
-  fi
-  # gmp_xoptcflags: First is best, one has to work.
-  eval xoptcflags=\$gmp_xoptcflags_$CC
-  if test -n "$xoptcflags"; then
-    gmp_found="no"
-    for xopt in $xoptcflags; do
-      CFLAGS_save="$CFLAGS"
-      CFLAGS="$CFLAGS $xopt"
-      AC_MSG_CHECKING([whether $CC accepts $xopt])
-      AC_LANG_C
-      AC_TRY_COMPILER([int main(){return(0);}], optok, cross)
-      if test "$optok" = "yes"; then
-        AC_MSG_RESULT([yes])
-        gmp_found="yes"
-        break
-      else
-        AC_MSG_RESULT([no])
-        CFLAGS="$CFLAGS_save"
-      fi
-    done
-    if test "$gmp_found" = "no"; then
-      echo ["$0: fatal: need a compiler that understands one of $xoptcflags"]
-      exit 1
-    fi
-  fi
 fi
 
 # Restore CFLAGS given on command line.
--- gmp-3.1.1/acinclude.m4.orig	Fri Aug 10 21:15:48 2001
+++ gmp-3.1.1/acinclude.m4	Fri Aug 10 21:29:50 2001
@@ -183,8 +183,6 @@
 dnl  Cache the result as `ac_cv_prog_CC'.
 AC_DEFUN(GMP_PROG_CC_SELECT,
 [AC_BEFORE([$0], [CC_PROG_CPP])
-AC_PROG_CC_WORKS
-AC_PROG_CC_GNU
 
 if test "$ac_cv_prog_gcc" = "yes"; then
   GCC=yes
@@ -766,7 +764,7 @@
 AC_MSG_CHECKING([if the C compiler ($CC) works with flags $CFLAGS])
 
 # Simple test for all targets.
-AC_TRY_COMPILER([int main(){return(0);}],
+AC_TRY_COMPILE([],[int main(){return(0);}],
                 tmp_works, tmp_cross)
 
 # Target specific tests.
@@ -830,6 +830,3 @@
 ])
 
 
-dnl  Deal with bad synchronization of Autoconf with Libtool.
-AC_DEFUN(AC_CANONICAL_BUILD, [_AC_CANONICAL_BUILD])
-AC_DEFUN(AC_CHECK_TOOL_PREFIX, [_AC_CHECK_TOOL_PREFIX])
