diff -ur gmp-2.0.2.old/Makefile.in gmp-2.0.2/Makefile.in
--- gmp-2.0.2.old/Makefile.in	Wed May 19 07:44:44 1999
+++ gmp-2.0.2/Makefile.in	Wed May 19 07:49:51 1999
@@ -26,6 +26,7 @@
 libdir = $(exec_prefix)/lib
 infodir = $(prefix)/info
 includedir = $(prefix)/include
+DESTDIR = 
 
 MV = mv
 CC = gcc
@@ -170,23 +171,23 @@
 
 install-strip: install-normal
 install-normal: installdirs libgmp.a gmp.info install-info-files
-	$(INSTALL_DATA) libgmp.a $(libdir)/libgmp.a
-	-chmod a-x $(libdir)/libgmp.a
-	$(INSTALL_DATA) $(srcdir)/gmp.h $(includedir)/gmp.h
-	-chmod a-x $(includedir)/gmp.h
-	$(INSTALL_DATA) libgmp.so.2.0.2 $(libdir)/libgmp.so.2.0.2
-	cd $(libdir);rm -f libgmp.so libgmp.so.2 libgmp.so.2.0
-	cd $(libdir);ln -s libgmp.so.2.0.2 libgmp.so.2
-	cd $(libdir);ln -s libgmp.so.2 libgmp.so
+	$(INSTALL_DATA) libgmp.a $(DESTDIR)$(libdir)/libgmp.a
+	-chmod a-x $(DESTDIR)$(libdir)/libgmp.a
+	$(INSTALL_DATA) $(srcdir)/gmp.h $(DESTDIR)$(includedir)/gmp.h
+	-chmod a-x $(DESTDIR)$(includedir)/gmp.h
+	$(INSTALL_DATA) libgmp.so.2.0.2 $(DESTDIR)$(libdir)/libgmp.so.2.0.2
+	cd $(DESTDIR)$(libdir);rm -f libgmp.so libgmp.so.2 libgmp.so.2.0
+	cd $(DESTDIR)$(libdir);ln -s libgmp.so.2.0.2 libgmp.so.2
+	cd $(DESTDIR)$(libdir);ln -s libgmp.so.2 libgmp.so
 install-bsdmp: installdirs libmp.a gmp.info install-info-files
-	$(INSTALL_DATA) libmp.a $(libdir)/libmp.a
-	-chmod a-x $(libdir)/libmp.a
-	$(INSTALL_DATA) $(srcdir)/mp.h $(includedir)/mp.h
-	-chmod a-x $(includedir)/mp.h
+	$(INSTALL_DATA) libmp.a $(DESTDIR)$(libdir)/libmp.a
+	-chmod a-x $(DESTDIR)$(libdir)/libmp.a
+	$(INSTALL_DATA) $(srcdir)/mp.h $(DESTDIR)$(includedir)/mp.h
+	-chmod a-x $(DESTDIR)$(includedir)/mp.h
 install-info-files: installdirs $(srcdir)/gmp.info
 	cd $(srcdir); for f in gmp.info*; \
-	do $(INSTALL_DATA) $$f $(infodir)/$$f; done
-	-chmod a-x $(infodir)/gmp.info*
+	do $(INSTALL_DATA) $$f $(DESTDIR)$(infodir)/$$f; done
+	-chmod a-x $(DESTDIR)$(infodir)/gmp.info*
 	# Attempt to edit the info directory node
 # Do this from the rpm spec file.
 #	if $(SHELL) -c 'install-info --version' >/dev/null 2>&1; then \
@@ -194,14 +195,14 @@
 #	  else true; fi
 
 installdirs: $(srcdir)/mkinstalldirs
-	$(srcdir)/mkinstalldirs $(includedir) $(libdir) $(infodir)
+	$(srcdir)/mkinstalldirs $(DESTDIR)$(includedir) $(DESTDIR)$(libdir) $(DESTDIR)$(infodir)
 
 uninstall:
-	rm -f $(libdir)/libgmp.a
-	rm -f $(includedir)/gmp.h
-	rm -f $(libdir)/libmp.a
-	rm -f $(includedir)/mp.h
-	rm -f $(infodir)/gmp.info*
+	rm -f $(DESTDIR)$(libdir)/libgmp.a
+	rm -f $(DESTDIR)$(includedir)/gmp.h
+	rm -f $(DESTDIR)$(libdir)/libmp.a
+	rm -f $(DESTDIR)$(includedir)/mp.h
+	rm -f $(DESTDIR)$(infodir)/gmp.info*
 
 clean mostlyclean:
 	rm -f *.o libgmp.a libmp.a gmp.dvi gmp.ps tmp.* tmp-*
