--- gmp-5.0.1/gmp-h.in.orig	2010-02-06 13:43:13.000000000 +0100
+++ gmp-5.0.1/gmp-h.in	2010-03-26 08:29:16.966713597 +0100
@@ -30,7 +30,12 @@
 #if ! defined (__GMP_WITHIN_CONFIGURE)
 #define __GMP_HAVE_HOST_CPU_FAMILY_power   @HAVE_HOST_CPU_FAMILY_power@
 #define __GMP_HAVE_HOST_CPU_FAMILY_powerpc @HAVE_HOST_CPU_FAMILY_powerpc@
-#define GMP_LIMB_BITS                      @GMP_LIMB_BITS@
+/* ugly hack: limb size differs depending on ABI; note: x32 ABI uses longlong limb, so also 64 bits */
+#if defined(__arch64__) || defined(__powerpc64__) || defined(__s390x__) || defined(__x86_64__) || defined(_LP64)
+#define GMP_LIMB_BITS                      64
+#else
+#define GMP_LIMB_BITS                      32
+#endif
 #define GMP_NAIL_BITS                      @GMP_NAIL_BITS@
 #endif
 #define GMP_NUMB_BITS     (GMP_LIMB_BITS - GMP_NAIL_BITS)
